pyk/src/pyk/ktool/krun.py

```diff
@@ -129,8 +133,14 @@ def run(
             pipe_stderr=pipe_stderr,
             bug_report=bug_report,
             debugger=debugger,
+            proof_hint=proof_hint,
         )
 
+        if proof_hint:
+            # Print the binary proof hint to stdout regardless of the output option
+            write(1, result.stdout)
```

Comment:
I'm not sure what the intent of the code here is; it seems confusing that the proof hint mode will unconditionally escape the Python framework and dump directly to stdout.

Reply:
I agree, it would be better to expose this operation mode as a separate function.

Reply:
This is the actual behavior of the flag in the krun. `kprint` can't handle this kind of binary, so the idea is just to dump it in the console as `krun` does. The alternative, from my point of view, would be to Dump into a file but escape the Python framework anyway.

Reply:
For instance, krun only `cat .../result.kore` when `--prof-hint` is passed as opposite to call `kore-print ... result.kore` when this flag isn't in use. 

So I believe this mimics the same behavior we already have.

Reply:
> The alternative, from my point of view, would be to Dump into a file but escape the Python framework anyway.

The point is that we manage `stdout` and `stderr` through the `CompletedProcess` object. If the hint has been captured by the subprocess library, then the way to get it to the calling app's `stdout` is not to bypass that management entirely; you should probably instead have the calling code write it out to a file yourself.